datadog_checks_base/datadog_checks/py.typed

```diff
@@ -0,0 +1 @@
+partial
```

Comment:
What does this do?

Reply:
Q: "Why `py.typed`?"
A: [PEP 561](https://www.python.org/dev/peps/pep-0561) requires that packages that have type annotations should have this marker `py.typed` file. Eg [HTTPX has it](https://github.com/encode/httpx/tree/master/httpx). Kinda ugly, but that's the spec, so let's roll with it.

Q: "Why `partial`?"
A: [PEP 561 motivates `partial` like so](https://www.python.org/dev/peps/pep-0561/#partial-stub-packages): "Many stub packages will only have part of the type interface for libraries completed, especially initially. or the benefit of type checking and code editors, packages can be `partial`." If we didn't use `partial` here then imports from `datadog_checks.base` would fail from mypy's perspective because it'd try to import stuff that doesn't have annotations (i.e. basically anything outside of `checks/base.py`). Here we tell it "only keep components that have valid type hints".